### PR TITLE
Textbox and Code Component Blur/Focus Fixes

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
- Re-focus textbox after submit
- Add focus and blur events to code component

Closes: #6303 & #6233

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/textbox/shared/Textbox.svelte [hunk 1]
```diff
@@ -68,6 +68,7 @@
 			scroll();
 		}
 		value_is_output = false;
+		el.focus();
```
Comment (abidlabs) [inline js/textbox/shared/Textbox.svelte:71]: Should this be at line 114 instead? Right now, it causes any change in the textbox value to focus on the textbox. Which means, e.g. if you launch 

```py
import gradio as gr

gr.Interface(lambda x:x, "textbox", "textbox").launch()
```

it focuses on the input textbox, which is not desired

**Comment by abidlabs** [comment]
I'm not able to get the `.blur()` and `.focus()` events to fire. Here's my code:

```py
import gradio as gr

def blur():
    print("blur")
    
def focus():
    print("focus")
    
def change():
    print("change")
    
with gr.Blocks() as demo:
    c = gr.Code()
    c.blur(blur, None, None)
    c.focus(focus, None, None)
    c.change(change, None, None)
    
demo.launch()
```

Note that `.change()` fires fine -- am I missing something?

**Comment by abidlabs** [comment]
Will take a look! Maybe we add a unit test for the blurring behavior?

**Comment by abidlabs** [comment]
Tested with the same demo as above, couple of issues:

* `blur()` event fires when the demo is first loaded
* the `focus()` event fires every time I type a character into the `Code` component
* In my tests, #6303 is not fixed.

I just tested:

```py
import gradio as gr

def echo(message, history):
    return message

demo = gr.ChatInterface(fn=echo, examples=["hello", "hola", "merhaba"], title="Echo Bot")
demo.launch()
```

after submitting, the textbox does not get re-focused

**Comment by dawoodkhan82** [comment]
Not sure what caused the regression for the textbox re-focus issue, but this pr should fix it. Also the code component blur and focus should work as expected now, although sometimes it misfires for me when testing. Let me know if you notice the same @abidlabs

**Comment by abidlabs** [comment]
LGTM working well for me @dawoodkhan82!

**Comment by VigneshHexo** [comment]
> LGTM working well for me @dawoodkhan82!

Could you please let me know, how can I accomplish the same re-focus feature while building Custom Chatbot with Gradio blocks? For example, here is a sample code snippet

```python
import gradio as gr
import random
import time

with gr.Blocks() as demo:
    chatbot = gr.Chatbot()
    msg = gr.Textbox()
    clear = gr.ClearButton([msg, chatbot])

    def respond(message, chat_history):
        bot_message = random.choice(["How are you?", "I love you", "I'm very hungry"])
        chat_history.append((message, bot_message))
        time.sleep(2)
        return "", chat_history
    
    msg.submit(respond, [msg, chatbot], [msg, chatbot])

if __name__ == "__main__":
    demo.queue()
    demo.launch(show_api=False, share=False, inline=False)
```

**Pull request merged**